src/components/data/DataTable.tsx

```diff
@@ -59,6 +65,25 @@ type ComponentProps<Row> = {
    */
   onConfirmRow?: (r: Row) => void;
 
+  /** Currently active order */
+  order?: Order<keyof Row>;
+
+  /**
+   * Callback when active order changes, by clicking a column header.
+   * When a header is clicked, if that's not the active order, it is set with
+   * order='ascending'.
+   * If the active header is clicked consecutively, direction rotates from
+   * 'ascending' to 'descending', and from 'descending' to no-active-order.
+   */
+  onOrderChange?: (order: Order<keyof Row>) => void;
+
+  /**
+   * Columns that can be used to order the table. Ignored if `onOrderChange` is
+   * not provided.
+   * Defaults to all columns.
+   */
+  orderableColumns?: Array<keyof Row>;
```

Comment:
The caller will need to provide code for every column they want to support sorting by. For that reason it might be better to require the caller to list all of the sortable columns instead of defaulting to assuming they are all sortable.

Reply:
Yep, I think you are right.